src/recordhost.cpp

```diff
@@ -140,12 +140,35 @@ RecordHost::PerfCapabilities fetchLocalPerfCapabilities(const QString& perfPath)
 
     return capabilities;
 }
+
+RecordHost::PerfCapabilities fetchRemotePerfCapabilities(const RemoteDevice& device)
+{
+    RecordHost::PerfCapabilities capabilities;
+
+    const auto buildOptions =
+        device.getProgramOutput({QStringLiteral("perf"), QStringLiteral("version"), QStringLiteral("--build-options")});
+    const auto help = device.getProgramOutput({QStringLiteral("perf"), QStringLiteral("--help")});
+
+    capabilities.canCompress = Zstd_FOUND && buildOptions.contains("zszd: [ on  ]");
+    capabilities.canSwitchEvents = help.contains("--switch-events");
+    capabilities.canSampleCpu = help.contains("--sample-cpu");
+
+    // TODO: implement
+    capabilities.canProfileOffCpu = false;
+    capabilities.privilegesAlreadyElevated = false;
+
+    capabilities.canUseAio = false; // AIO doesn't work with perf streaming
+    capabilities.canElevatePrivileges = false; // we currently don't support this
+
+    return capabilities;
+}
 }
 
 RecordHost::RecordHost(QObject* parent)
     : QObject(parent)
     , m_checkPerfCapabilitiesJob(this)
     , m_checkPerfInstalledJob(this)
+    , m_remoteDevice(this)
```

Comment:
passing `this` when this is owned on the stack sounds wrong. make this a ptr and forward declare in the header? and only actually instantiate when we are _not_ local?